### PR TITLE
Only set mock location helper for real device

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Addresses https://github.com/appium/appium/issues/8487

### CODE REVIEW

File: lib/android-helpers.js [hunk 1]
```diff
@@ -467,11 +467,11 @@ helpers.initDevice = async function (adb, opts) {
   }
   if (!opts.avd) {
```
Comment (vrunoa) [inline lib/android-helpers.js]: @mykola-mokhnach would it be better to call `isEmulator` ?

Comment (mykola-mokhnach) [inline lib/android-helpers.js]: no, helpers don't have access to driver context

Comment (vrunoa) [inline lib/android-helpers.js]: we seen `opts.avd` fail with @imurchie,  since you can avoid setting the avd capability if the emulator is already running

Comment (mykola-mokhnach) [inline lib/android-helpers.js]: then it should be another bug. We have to synchronize **avd** capability value in such case

Comment (jlipps) [inline lib/android-helpers.js]: > no, helpers don't have access to driver context

we could pass in the result of `isEmulator()` to this function as a param though...

**Pull request closed without merging**